### PR TITLE
[New Rule] Adding Coverage for `AWS SSM Command Document Created by Rare User`

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Pull Request

*Issue link(s)*:
* https://github.com/elastic/ia-trade-team/issues/456
* https://github.com/elastic/ia-trade-team/issues/272

## Summary - What I changed
Identifies when an AWS Systems Manager (SSM) command document is created by a user who does not typically perform this action. Adversaries may create SSM command documents to execute commands on managed instances, potentially leading to unauthorized access, command and control, data exfiltration and more.

This rule specifies that the document type is of [command](https://docs.aws.amazon.com/systems-manager/latest/userguide/documents.html), which is how malicous scripts or commands can be executed on other compute instances regardless of VPC placement.

_Run Command, a capability of AWS Systems Manager, uses Command documents to run commands. State Manager, a capability of AWS Systems Manager, uses command documents to apply a configuration. These actions can be run on one or more targets at any point during the lifecycle of an instance. Maintenance Windows, a capability of AWS Systems Manager, uses Command documents to apply a configuration based on the specified schedule._

## How To Test
Screenshot 2024-11-01 at 1 39 20 PM

## Checklist

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

File: rules/integrations/aws/execution_ssm_command_document_created_by_rare_user.toml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+risk_score = 21
+rule_id = "50a2bdea-9876-11ef-89db-f661ea17fbcd"
+severity = "low"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: AWS SNS",
+    "Data Source: AWS Systems Manager",
+    "Resource: Investigation Guide",
```
Comment (terrancedejesus) [inline rules/integrations/aws/execution_ssm_command_document_created_by_rare_user.toml:69]:
```suggestion
    "Resources: Investigation Guide",
```

**Review by Aegrah — APPROVED** [review]

File: rules/integrations/aws/execution_ssm_command_document_created_by_rare_user.toml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+event.dataset: "aws.cloudtrail"
+    and event.provider: "ssm.amazonaws.com"
+    and event.action: "CreateDocument"
+    and event.outcome: "success"
+    and aws.cloudtrail.response_elements: *documentType=Command*
```
Comment (imays11) [inline rules/integrations/aws/execution_ssm_command_document_created_by_rare_user.toml:81]: why this distinction?

Comment (terrancedejesus) [reply rules/integrations/aws/execution_ssm_command_document_created_by_rare_user.toml:81]: https://docs.aws.amazon.com/systems-manager/latest/userguide/documents.html

Command documents are the focus here to capture RunShellScript RunPowerShellScript or custom documents that are meant to run system commands upon detonation.

**Review by shashank-elastic — APPROVED** [review]

**Review by imays11 — APPROVED** [review]

**Pull request merged**